test/views/git-tab-view.test.js

```diff
@@ -282,4 +282,13 @@ describe('GitTabView', function() {
     wrapper.instance().focusAndSelectRecentCommit();
     assert.isTrue(setFocus.calledWith(GitTabView.focus.RECENT_COMMIT));
   });
+
+  it('calls changeProjectWorkingDirectory when a project is selected', async function() {
+    const select = sinon.spy();
+    const path = 'test/path';
+    const wrapper = mount(await buildApp({changeProjectWorkingDirectory: select}));
+    wrapper.find('.github-Project-path.input-select').simulate('change', {target: {value: path}});
+    assert.isTrue(select.calledWith(path));
+    wrapper.unmount();
```

Comment:
This is a bit different than the way the rest of the suites are structured. This is more of an integration test: it uses `mount` and simulates DOM events on selectors that are controlled by child components. Generally, we use `shallow` as much as we can, and we focus tests in each suite on the markup and immediate children created by the component under test in each file.

Ideally, this test would be in a new suite dedicated to the `HeaderView` component. We _could_ have tests up the component hierarchy to ensure the callbacks are passed properly, but it likely wouldn't be worth it 😄

Reply:
The only reason for this test was branch coverage. So what should I do remove it or move it to header-view.test.js?

Reply:
Oh, I see, you need to cover this line?

https://github.com/atom/github/blob/5bb76cb7f003ec5ec3ee930e76caf253c5e19406/lib/views/git-tab-view.js#L121

Here's how I would do that in this suite:

```js
it('calls changeWorkingDirectory when a project is selected', async function() {
  const changeWorkingDirectory = sinon.spy();
  const wrapper = shallow(await buildApp({changeWorkingDirectory}));
  wrapper.find('HeaderView').prop('handleProjectSelect')({target: {value: 'some-path'}});
  assert.strictEqual(changeWorkingDirectory.calledWith('some-path'));
});
```

The logic here is that if we keep the `git-tab-view.test.js` tests written in terms of things defined directly in `GitTabView`, we can make changes further down the tree, like changing the markup inside `HeaderView`, without needing to change this suite, too. (And yeah, I know there are a bunch of existing tests that violate this 😬 It's one of those systemic changes I alluded to before. For now we just try to keep new tests written in this style.)

Reply:
👍